lsp/server/src/server.ts

```diff
@@ -155,6 +146,58 @@ connection.languages.diagnostics.on(async (params) => {
     }
 });
 
+connection.onCodeAction((params) => {
+    const textDocument = documentCache.get(params.textDocument.uri);
+    const diagnostics = params.context.diagnostics;
+	if (textDocument === undefined || diagnostics.length === 0) {
+		return undefined;
+	}
+    
+	const result: CodeAction[] = [];
+    diagnostics.forEach((item) => {
+        const { data } = item;
+        if (data !== undefined) {
+            const metData = data as DiagnosticMetaData;
+            if (metData.id !== undefined) {
+
+                const suppressedIds = new Set(settings.diagnostic.suppressedIds);
```

Comment:
generate the quickfix to suppress this diagnostic producer.

Reply:
A `Set` created from suppressed Ids don't need to be created in each loop of `forEach` does it?

Reply:
when instantiate a new set, a array like structure could be passed in as constructor parameter.  the items from input parameter will become items of the set.  so no need to loop :)